### DISCUSSION TITLE
Rests in Music

### DISCUSSION THREAD

**Discussion by estebanc08** [w3c-cg/musicxml#488] [Q&A]
I was wondering if anyone could help me with how to get position of a rest using XML. I have a project on my account that reads in an XML file and tries to output the sheet music as both audio and visual. I am using the x-pos to get the location of the notes in each measure instead of using the note type as I found it easier for now, but I have no clue how to transfer this over for the rest. https://youtu.be/-bjAls0SSqA . As of right now, the note durations are all hard coded in and there are no dynamics so it sounds kinda bad, but I want to make the rests work before I move on and am not sure how to calculate where in a measure a rest is.

**Comment by mscuthbert** [comment]
I'm a little confused about the question, and I'm afraid that the video doesn't help -- musicxml defines the semantic (understood conceptually) position of a note or a rest by its number of divisions, and it defines its exact placement on the page by the optional absolute-x, absolute-y positions.  Are you sure you're thinking about the musicxml standard or something like MIDI?

**Reply by estebanc08** [reply]
Hello, I was trying something different but found it just would not reasonably work. Ended up doing Midi after all, thanks though. I am a college student just trying new things. Havent worked on it in a while but this is what I ended up with: https://www.youtube.com/watch?v=kBymkkDfmkU
